core/tests/protractor_utils/StoryEditorPage.js

```diff
@@ -158,6 +162,27 @@ var StoryEditorPage = function() {
     await general.scrollToTop();
   };
 
+  this.navigateToChapterWithName = async function(chapterName) {
+    await waitFor.visibilityOf(
+      await chapterTitles.first(), 'Chapter list taking too long to appear');
+    var chapterIndex = -1;
+    for (var i = 0; i < await chapterTitles.count(); i++) {
+      if (await chapterTitles.get(i).getText() === chapterName) {
+        chapterIndex = i;
+        break;
+      }
+    }
+    expect(chapterIndex).not.toEqual(-1);
+
+    await action.click('Chapter list item', chapterTitles.get(i));
+    await waitFor.pageToFullyLoad();
```

Comment:
Is a loading symbol shown as the page loads? If not, this line won't do anything.

Reply:
Removed. Added waitFor for an element on that page.
Done.